tests/sentry/api/endpoints/test_project_templates_index.py

```diff
@@ -55,3 +58,70 @@ def test_get__no_access(self):
         # Ensure this errors with 403, as the user does not have access to the organization
         response = self.get_error_response(org_two.id, status_code=403)
         assert response.status_code == 403
+
+
+class ProjectTemplateIndexPostTest(APITestCase):
+    endpoint = "sentry-api-0-organization-project-templates"
+
+    def setUp(self):
+        super().setUp()
+        self.user = self.create_user()
+        self.organization = self.create_organization(owner=self.user)
+        self.team = self.create_team(members=[self.user])
+
+        self.login_as(self.user)
+
+        self.project_template = self.create_project_template(organization=self.organization)
+        self.project_template_two = self.create_project_template(organization=self.organization)
+
+    def test_post__no_feature(self):
+        response = self.get_error_response(self.organization.id, method="POST", status_code=404)
+        assert response.status_code == 404
+
+    @with_feature(PROJECT_TEMPLATE_FEATURE_FLAG)
+    def test_post__as_member(self):
+        org_two = self.create_organization()
+        self.create_team(organization=org_two, members=[self.user])
+        self.create_project_template(organization=org_two)
+
+        response = self.get_error_response(org_two.id, method="POST", status_code=403)
+        assert response.status_code == 403
+
+    @with_feature(PROJECT_TEMPLATE_FEATURE_FLAG)
+    def test_post(self):
+        response = self.get_success_response(
+            self.organization.id, method="POST", name="Test Project Template"
+        )
+        assert response.status_code == 201
+
+    @with_feature(PROJECT_TEMPLATE_FEATURE_FLAG)
+    def test_post__with_options(self):
+        test_options = {"test-key": "value"}
+        response = self.get_success_response(
+            self.organization.id,
+            method="POST",
+            name="Test Project Template",
+            options=test_options,
+        )
+
+        assert response.status_code == 201
+        new_template = ProjectTemplate.objects.get(id=response.data["id"])
+        created_options = {opt.key: opt.value for opt in new_template.options.all()}
+        assert created_options == test_options
+
+    @with_feature(PROJECT_TEMPLATE_FEATURE_FLAG)
+    def test_post__no_name(self):
+        response = self.get_error_response(self.organization.id, method="POST", status_code=400)
```

Comment:
what kinds of names do we expect people to come up with for these? could it possibly benefit from autogenerated names?

Reply:
there's a test helper that creates and uses auto generated names. so basically everything that uses `self.create_template` kinda has that baked in 🎉